Initialize ESP32 Test projects :)

antholuo opened the pull request:
Adds std and no_std test project.

Seems like we'll be going the no_std route since that's the way the community is moving (esp-hal is now the primary focus).

Uprev'd everything to be able to accept esp-hal 1.0.0-beta.1. We'll continue working with this ESP-HAL as they push releases. Exciting times :) 

Note that the C6 doesn't have 4 timers, only 1 timer per group :( (at least for now?). S3 might be a better option but I wouldn't know.

Will squash before merge.
https://cdn.discordapp.com/attachments/1382883730531553361/1383173777566994432/IMG_1531.mov?ex=684dd476&is=684c82f6&hm=4462493ace6e3b3a94c75823a58c5f0eac8b0b5dc8167c91b4999a70b138e094&

antholuo committed:
Test Project Init, no_std WS2812 LED cycling active

HardyYu reviewed (approved):
I don't have too many comments on this pr. It's a good baseline to start with.

antholuo committed:
LP Core functional - using ESP-HAL project

antholuo committed:
Add our own LP_Core (no more using esp-hal's example!)

Had a fair number of problems fighting the compiler and the linker. We gucci gang now.

antholuo committed:
Gitignore
